### PR TITLE
State hierarchy

### PR DESCRIPTION
**Pull request by hackaugusto**
This PR introduces the state objects used to represent the registry / channel manager and an umbrela state object to hold the whole app state.

### CODE REVIEW

File: raiden/transfer/state.py [hunk 1]
```diff
@@ -23,6 +25,231 @@ def balanceproof_from_envelope(envelope_message):
...
+    def __init__(
+            self,
+            address: typing.address,
+            token_network_list: typing.List['TokenNetworkState']):
+
+        if not isinstance(address, typing.address):
+            raise ValueError('address must be an address instance')
+
+        self.address = address
+        self.tokensidentifiers_to_tokennetworks = {
```
Comment (palango) [inline raiden/transfer/state.py]: Should that be `tokenidentifiers`, so without the 's' in the middle?

File: raiden/transfer/state_change.py [hunk 2]
```diff
@@ -121,6 +127,186 @@ def __str__(self):
         )
 
 
+class ActionInitNode(StateChange):
+    def __init__(self, block_number: int):
```
Comment (palango) [inline raiden/transfer/state_change.py]: Should block_number be of type `typing.block_number`?

File: raiden/transfer/state.py [hunk 3]
```diff
@@ -23,6 +25,231 @@ def balanceproof_from_envelope(envelope_message):
...
+        if not isinstance(block_number, typing.block_number):
+            raise ValueError('block_number must be an block_number')
+
+        self.block_number = block_number
+        self.identifiers_to_paymentnetworks = dict()
+        self.nodeaddresses_to_networkstates = dict()
+        self.payment_mapping = PaymentMappingState()
+
+    def __repr__(self):
+        return '<NodeState block:{} networks:{} qtd_transfers:{}>'.format(
```
Comment (palango) [inline raiden/transfer/state.py]: What does `qtd` stand for?

**Comment by LefterisJP** [comment]
I am in the process of rebasing this PR to latest master. It's hell. Just noting it here so that nobody else steps on my toes.

**Comment by LefterisJP** [comment]
Rebase is complete

File: raiden/transfer/state.py (hunk 3 above)
Comment (LefterisJP) [reply raiden/transfer/state.py]: Good questin. I also wonder. Maybe he wanted to say quantity? so `qty`? @hackaugusto ?

File: raiden/transfer/state.py (hunk 1 above)
Comment (LefterisJP) [reply raiden/transfer/state.py]: it is already. Did you change that yourself @palango ?

Comment (palango) [reply raiden/transfer/state.py]: nope

File: raiden/transfer/state.py (hunk 3 above)
Comment (palango) [reply raiden/transfer/state.py]: Quantity makes sense

File: raiden/transfer/state.py (hunk 1 above)
Comment (palango) [reply raiden/transfer/state.py]: But I still see the s in the middle

Comment (LefterisJP) [reply raiden/transfer/state.py]: aaah lol after reading it 3 times I now understand what you meant

Comment (palango) [reply raiden/transfer/state.py]: haha, I have to work on my writing then ;)

File: raiden/transfer/state.py (hunk 3 above)
Comment (hackaugusto) [reply raiden/transfer/state.py]: Yep, it's quantity. `qtd` is the portuguese abbreviation

Comment (LefterisJP) [reply raiden/transfer/state.py]: ha nice! So `quantidade` -> `qtd`. Should rename it to `qty` for clarity eventually ;)

**Pull request merged**